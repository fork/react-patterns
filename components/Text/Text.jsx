import styled from 'styled-components';

import { color } from '../../stylesheets';

const Text = styled.p`
  color: ${props => (props.color ? color(props.color)(props) : 'inherit')};
  text-align: ${props => props.alignment};
  text-transform: ${props => props.transform};
  margin: 0;
`;

export default Text;
