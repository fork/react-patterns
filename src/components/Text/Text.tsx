import { ReactNode } from 'react';
import styled from 'styled-components';

import { color, typography } from '../../stylesheets';
import { Colors, Typography } from '../../tokens';

export type TextProps = {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span' | 'figure' | 'div' | 'cite';
  variant?: Typography;
  color?: Colors;
  children: ReactNode;
};

const Text = styled.p<TextProps>`
  color: ${props => (props.color ? color(props.color)(props) : 'inherit')};
  margin: 0;

  ${props => typography(props.variant || 'copy')(props)}
`;

export default Text;
