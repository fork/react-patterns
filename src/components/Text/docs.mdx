import { Story, Canvas } from '@storybook/addon-docs/blocks';

# Text

<Canvas>
  <Story id="tokens-text--headline-1" />
</Canvas>

<Canvas>
  <Story id="tokens-text--headline-2" />
</Canvas>

<Canvas>
  <Story id="tokens-text--headline-3" />
</Canvas>

<Canvas>
  <Story id="tokens-text--headline-4" />
</Canvas>

<Canvas>
  <Story id="tokens-text--copy" />
</Canvas>

<Canvas>
  <Story id="tokens-text--copy-small" />
</Canvas>

<Canvas>
  <Story id="tokens-text--quote" />
</Canvas>

<Canvas>
  <Story id="tokens-text--button" />
</Canvas>

<Canvas>
  <Story id="tokens-text--link" />
</Canvas>

<Canvas>
  <Story id="tokens-text--form-value" />
</Canvas>

<Canvas>
  <Story id="tokens-text--form-label" />
</Canvas>

<Canvas>
  <Story id="tokens-text--form-hint" />
</Canvas>
