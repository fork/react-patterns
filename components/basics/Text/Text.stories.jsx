import React from 'react';

import Text from './Text';

export default {
  title: 'Basics/Text',
  component: Text
};

export const Normal = () => <Text>This is a text component</Text>;
export const Colored = () => <Text color="primary">This is a text component</Text>;
