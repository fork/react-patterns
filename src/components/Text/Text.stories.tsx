import React from 'react';
import { Story, Meta } from '@storybook/react';

import Text, { TextProps } from './Text';

export default {
  title: 'Components/Text',
  component: Text
} as Meta;

const Template: Story<TextProps> = ({ children, ...args }) => <Text {...args}>{children}</Text>;

export const Main = Template.bind({});
Main.args = {
  children: 'This is a text component',
  variant: 'main'
};

export const Sub = Template.bind({});
Sub.args = {
  ...Main.args,
  variant: 'sub'
};

export const Colored = Template.bind({});
Colored.args = {
  ...Main.args,
  color: 'primary'
};
