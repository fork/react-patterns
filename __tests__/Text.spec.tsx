import React from 'react';
import { shallow } from 'enzyme';

import Text from '../src/components/Text';

describe('Text', () => {
  test('Text should render correctly', () => {
    const component = shallow(<Text>Primary Text</Text>);

    expect(component).toMatchSnapshot();
  });
});
