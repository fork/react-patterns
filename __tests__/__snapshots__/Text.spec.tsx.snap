// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Text should render correctly 1`] = `
<p
  className="sc-bdnxRM dSnbYG"
>
  Primary Text
</p>
`;
